### PR TITLE
[EventHubs]: Deprecate message retention in days

### PR DESCRIPTION
**Pull request by damodaravadhani**
## Description

Added breaking change description for parameters

## Checklist

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/EventHub/EventHub.Autorest/README.md [hunk 1]
```diff
@@ -343,6 +343,16 @@ directive:
...
       property-name: MessageRetentionInDays
 
+  - where:
+      verb: New
+      subject: EventHub
+      parameter-name: MessageRetentionInDays
+    set:
+      breaking-change:
+        change-description: -MessageRetentionInDays would be deprecated and would be replaced by -RetentionTimeInHours
+        deprecated-by-version: 9.5.0
```
Comment (isra-fel) [inline src/EventHub/EventHub.Autorest/README.md:353]: Here should be the next expected major version of Az.EventHub i.e. 4.0.0

File: src/EventHub/EventHub.Autorest/README.md [hunk 2]
```diff
@@ -343,6 +343,16 @@ directive:
...
+        change-effective-date: 2023/03/07
```
Comment (isra-fel) [inline src/EventHub/EventHub.Autorest/README.md:354]: Next breaking change window is around next Micorosft Build but not finalized. I suggest do not include a effective date.

**Review by isra-fel — CHANGES_REQUESTED** [review]

File: src/EventHub/EventHub.Autorest/README.md [hunk 3]
```diff
@@ -343,6 +343,14 @@ directive:
     set:
       property-name: MessageRetentionInDays
 
+  - where:
+      verb: New
+      subject: EventHub
+      parameter-name: MessageRetentionInDays
+    set:
+      breaking-change:
+        change-description: -MessageRetentionInDays would be deprecated and would be replaced by -RetentionTimeInHours
```
Comment (BethanyZhou) [inline src/EventHub/EventHub.Autorest/README.md:352]: I think we can keep deprecated-by-version: 4.0.0. For Az.EventHub, we can expect a major version change

**Comment by damodaravadhani** [comment]
@BethanyZhou, addressed the comments here. Please review and merge.

File: src/EventHub/EventHub.Autorest/custom/Set-AzEventHub.ps1 [hunk 4]
```diff
@@ -89,7 +89,8 @@ function Set-AzEventHub{
         ${SkipEmptyArchive},
 
         [Parameter(HelpMessage = "Number of days to retain the events for this Event Hub, value should be 1 to 7 days")]
-		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
+		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("MessageRetentionInDays", "4.0.0", "", ChangeDescription = "-MessageRetentionInDays would be deprecated and would be replaced by -RetentionTimeInHours")]
+        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
```
Comment (BethanyZhou) [inline src/EventHub/EventHub.Autorest/custom/Set-AzEventHub.ps1:93]:
```suggestion
        	[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
```

File: src/EventHub/EventHub.Autorest/custom/Set-AzEventHub.ps1 [hunk 5]
```diff
@@ -89,7 +89,8 @@ function Set-AzEventHub{
         ${SkipEmptyArchive},
 
         [Parameter(HelpMessage = "Number of days to retain the events for this Event Hub, value should be 1 to 7 days")]
-		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
+		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("MessageRetentionInDays", "4.0.0", "", ChangeDescription = "-MessageRetentionInDays would be deprecated and would be replaced by -RetentionTimeInHours")]
+        	[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
 		[System.Int64]
 		# Number of days to retain the events for this Event Hub, value should be 1 to 7 days
 		${MessageRetentionInDays},
```
Comment (BethanyZhou) [inline src/EventHub/EventHub.Autorest/custom/Set-AzEventHub.ps1:96]:
```suggestion
		[Microsoft.Azure.PowerShell.Cmdlets.EventHub.Runtime.ParameterBreakingChangeAttribute("MessageRetentionInDays", "4.0.0", "", ChangeDescription = "-MessageRetentionInDays would be deprecated and would be replaced by -RetentionTimeInHours")]
        [Microsoft.Azure.PowerShell.Cmdlets.EventHub.Category('Body')]
	[System.Int64]
	# Number of days to retain the events for this Event Hub, value should be 1 to 7 days
	${MessageRetentionInDays},
```

**Review by BethanyZhou — APPROVED** [review]

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**